### PR TITLE
Create documentation for Argo Rollouts Plugin.

### PR DESCRIPTION
**Pull request by thisisnotashwin**
### Description

VERCEL LINK: https://consul-31z9s3g9y-hashicorp.vercel.app/consul/docs/k8s/deployment-configurations/argo-rollouts-configuration

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+spec:
+  selector:
+    app: test-service
+  ports:
+    - name: http
+      port: 80
+      targetPort: 8080
+```
+
+1. Create the Service Resolver and Service Splitter CRDs for your stable and canary services.
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:142]:
```suggestion
1. Create the Service Resolver and Service Splitter CRDs for your stable service. Argo will automate the modification of these CRDs during canary deployments.
```

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,221 @@
+---
+layout: docs
+page_title: Using Argo Rollouts for Canary upgrades with Consul on Kubernetes
+description: >-
+  Configuration for Argo Rollouts Controller to enable Canary deployments for subset based routing using Rollouts to integrate with Consul Service Mesh.
+---
+
+NOTES:
+**_1. Currently, the Consul Traffic Routing Rollouts Plugin only supports subset based routing._**
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:9]: Fix the newline here, showing up weird on vercel sight

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+### Install the RBAC
+
+After either mounting the binary or using an init container apply the RBAC using the provided `yaml/rbac.yaml`
+```bash
+kubectl apply -f https://raw.githubusercontent.com/argoproj-labs/rollouts-plugin-trafficrouter-consul/main/yaml/rbac.yaml
+```
+
+## Usage
+
+1. Create the Kubernetes Service that will be utilized by the service being rolled out. Additionally, configure any Service Defaults and Proxy Defaults required for the service.
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:127]: Using `1.` each time isn't working as intended when the site renders. I know markdown rendering usually handles this, but each numbered item is coming up as `1.`

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    - name: consul-route-plugin
+      mountPath: /plugin-bin/hashicorp
+```
+```bash
+helm install argo-rollouts argo/argo-rollouts -f values.yaml -n argo-rollouts
+```
+
+### Stand-alone installation
+
+NOTES:
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:69]: Fix newline, not rendering as expected on vercel preview

File: website/data/docs-nav-data.json [hunk 5]
```diff
@@ -1328,6 +1328,10 @@
             "title": "Consul Enterprise",
             "path": "k8s/deployment-configurations/consul-enterprise"
           },
+          {
+            "title": "Argo Rollouts Configuration",
```
Comment (david-yu) [inline website/data/docs-nav-data.json:1332]:
```suggestion
            "title": "Argo Rollouts Progressive Delivery",
```

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 6]
```diff
@@ -0,0 +1,219 @@
+---
+layout: docs
+page_title: Using Argo Rollouts for Canary upgrades with Consul on Kubernetes
```
Comment (david-yu) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:3]:
```suggestion
page_title: Argo Rollouts Progressive Delivery with Consul on Kubernetes
```

This title I believe seems to align with other Argo Rollouts documentation so proposing.

**Review by david-yu — APPROVED** [review]

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+-> **Note:** Currently, the Consul Traffic Routing Rollouts Plugin only supports subset based routing.
+
+### Install Rollouts Using Helm (Init Container)
+This is the preferred method for installing this plugin.
+
+Add the following code to your `values.yaml` file, then install the argo-rollouts by helm:
+
+```yaml
+  initContainers:
```
[the remark this answers is not in the document]
Comment (thisisnotashwin) [reply website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:31]: Good catch!! added the missing top level field.

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+This page describes the process to configure and use the [Argo Rollouts Controller](https://argo-rollouts.readthedocs.io/en/stable/) with Consul on Kubernetes to manage advanced subset-based routing for Canary deployments.
+
+Consul's support for Argo Rollouts is currently limited to subset-based routing.
+
+## Install Argo Rollouts Controller
+
+There are three methods for installing the Argo Rollouts Controller with Consul on Kubernetes:
+
+1. [Install with Helm and init containers](#install-rollouts-using-helm-init-containers). We recommend installing the Argo Rollouts Controllor using this method.
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:18]:
```suggestion
1. [Install Rollouts Using Helm and init containers](#install-rollouts-using-helm-and-binary). We recommend installing the Argo Rollouts Controllor using this method.
```

File: website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+1. [Build the binary and install with Helm](#install-rollouts-using-helm-binary)
```
Comment (wilkermichael) [inline website/content/docs/k8s/deployment-configurations/argo-rollouts-configuration.mdx:19]:
```suggestion
1. [Install Rollouts Using Helm and binary](#install-rollouts-using-helm-and-binary)
```

**Review by wilkermichael — APPROVED** [review]
Just adding some fixes to the links

**Pull request merged**